subsys/bluetooth/mesh/adv.c

```diff
@@ -41,8 +41,15 @@ const uint8_t bt_mesh_adv_type[BT_MESH_ADV_TYPES] = {
 
 K_FIFO_DEFINE(bt_mesh_adv_queue);
 
+static void adv_buf_destroy(struct net_buf *buf)
+{
+	struct bt_mesh_adv *send = BT_MESH_ADV(buf);
+
+	bt_mesh_adv_send_end(0, send->cb, send->cb_data);
+}
```

Comment:
The failing babblesim test suggests that we have a memory leak in the adv bufs somewhere. I think custom destroy callbacks must call `net_buf_destroy()` manually, as `net_buf_unref()` won't when we have custom destroy functions.